samples/gui/lvgl/boards/nrf52_pca10040.conf

```diff
@@ -0,0 +1,14 @@
+CONFIG_SPI=y
+CONFIG_SPI_1=y
+CONFIG_SPI_NRFX=y
+
+CONFIG_ILI9340=y
+CONFIG_ILI9340_RGB565=y
+CONFIG_ILI9340_RGB565_SWAP=y
```

Comment:
CONFIG_LVGL_COLOR_16_SWAP can be used to achieve the same functionality without the need of an extra buffer.

Reply:
When LittlevGL has RGB565 byte swap enabled, it becomes slower. On nRF52 devices, the LVGL full-screen processing time may be increased from 70ms to 100ms. On the other hand, the ILI9340 RGB565 byte swap just adds an overhead of 5ms.

Can we keep the ILI9340 RGB565 byte swap feature? Some users would like to boost the processing speed by giving a bit more RAM. It is also my preference.

Reply:
Pretty surprised the overhead of lvgl is that muche for a byte swap. Still I would move the functionality out of the driver, why would we increase the complexity of a driver to solve a library issue?

Further on lvgl "issue" it could be an option to define a ```zephyr_vdb_write``` function that takes care of the swap in [lib/gui/lvgl/lvgl_color_16.c](https://github.com/zephyrproject-rtos/zephyr/blob/master/lib/gui/lvgl/lvgl_color_16.c) instead of using the lvgl swap implementation.

Reply:
Let's handle LittlevGL RGB565 byte swap in another PR.

When CONFIG_LVGL_COLOR_16_SWAP is enabled, LittlevGL suffers a compilation error. Need to disable CONFIG_LVGL_OBJ_CANVAS to build the library.

Reply:
The patch is updated. Your comments are addressed. Please review. Thanks.